### ISSUE TITLE
Include this plugin in pibooth page

### ISSUE THREAD

**Issue by werdeil** [DJ-Dingo/pibooth-lcd-I2c#2]
Hi @DJ-Dingo 

I open a subject here to discuss the future of this plugin.

As you said in [https://github.com/pibooth/pibooth/issues/185](https://github.com/pibooth/pibooth/issues/185) let's make this plugin more generic and integrate it in the pibooth organisation repository.

Here are the things I think we can do:
- Change the plugin name to use lower case (pibooth-lcd-i2c for example) to be consistent with other plugins
- Update the README to be more like the one on [https://github.com/pibooth/pibooth-extra-lights](https://github.com/pibooth/pibooth-extra-lights)
- Add option in the config to allow some customization (as you said: "Text", "Port Expanders", and port "Address") 
- Add the setup.py part and then release a first version on pypi
- Handle other LCD configurations...
- Transfer the repository to the pibooth organisation.

I can do any of these, for some of them I can provide you pull requests, however some of these need rights on the repository, but if you give me access as a contributor to the project I can do it too (but I'm not forcing you).

**Comment by DJ-Dingo** [comment]
Hi @werdeil 

I have changed the name, and to lower case.
I've changed some of the readme file, but looking into make it more like extra-light :-)

**Comment by werdeil** [comment]
@DJ-Dingo for me the plugin is now stable enough to release a first version of pipy. If you want I can do it completely.

it will then appear in pipy and would be installable directly through `pip` like [https://pypi.org/project/pibooth-sound-effects/](https://pypi.org/project/pibooth-sound-effects/)

**Comment by DJ-Dingo** [comment]
Yes fine you can do that, but the new Rework config options is not working yet.

If you want to use the Refactoring version, i found out that you can´t use Free_Text_1 and Free_Text_2 at the same time
Free_Text_2 wont show up on line 2

Same if you want to use Free_Text_1 twice,  on line 1 and line 2
Free_Text_1 on line 2 don´t show up.

so maybe we should have the Rework config working first??

**Comment by werdeil** [comment]
Strange but yes, let's try to have the rework config working if the refactored one is not working correctly

**Comment by DJ-Dingo** [comment]
I'm also thinking of going through the readme file again, for any errors, or things that don't need to be there. And maybe also a new top image, with several lcd's which show some of the options in action.

I can still change in readme when it appears in pipy right?

**Comment by werdeil** [comment]
No the readme in pipy correspond to the one of the packaged version but we can still do a new release if necessary.

I've changed a bit the README (mainly formatting) to make it render correctly on pipy. We shall to use the html tags `</br>` but only skip lines instead

For me we are ready to do a first release except if you still want to improve the README, do you want it to be version 1.1.3? I would personally name it version 1.0.0 (first time the version is really released) but it is up to you, only the next versions will have to be upward.

**Comment by DJ-Dingo** [comment]
Fine with 1.0.0 - I just needed numbers to keep track of everything in my folder :-)

I got too tired yesterday, so did not get to go through the whole readme file. i'm not totally crazy about all the spaces between all the lines thats why i used html tags. But thatś just my eyes. :-)

Yes, there are a few things I just wanted to change. But think itś fine for now
Most impotant is that there are no tecnical errors. The look of the readme, is something i have to look at after it is released.
It is very simple to setup and lcd with an I2c and levelconverter (5 wires). And still the readme files guidance are long. But i did not wanted to leave things out. So maybe i will rewrite things later.

Cool what you made with the Date-time_formatcode file.

I really miss that it remembers text for every choice in menu.
So text follows your choice in the menu. Otherwise you have to write new text every time you choose something new. That is, if you choose **Printed**, and write Printed, and afterwards choose something else like **Forgotten** without changing the text, then text and counter no longer match each other. And that it is hard to remember the date-time format-code.

The good news is that you can now have text in line 3,4 on a 16x2 without any crash. So you could keep Date-Time format-codes there for later use, if you only have a 16x2 lcd.

The last thing that just needs to be changed in the py file is that type must come before text. So it's called
Line 1 Type
Line 1 Text

So are you gonna release it on Pypi, or should i ?

**Comment by werdeil** [comment]
> I really miss that it remembers text for every choice in menu.

I see your point but I don't think this will be such an issue, and it will be more visible with the type before the text (as you suggested).
I understand about the date format, it is indeed tricky to remember the previous format, but I don't think users will change it often, and the page you added is a good help to define the one you want.
But if you really find it worse we can still go back to a version with all lines defined in the config (but it will make a really big menu).

>So are you gonna release it on Pypi, or should I ?

I will just waiting for your go (I'll just push the order change in a few minutes)

**Comment by DJ-Dingo** [comment]
Okay Go.

But why will the menu get bigger.
It only needs to change text in the text field if you change type.
(requires it to save every time you change type if you have changed in text)

The config will be bigger, unless it can be done on 1 line

**Comment by werdeil** [comment]
[https://pypi.org/project/pibooth-lcd-i2c/1.0.0/](https://pypi.org/project/pibooth-lcd-i2c/1.0.0/) 👏 

> It only needs to change text in the text field if you change type.

Unfortunately It is not possible to update the menu in live for now, so you'll need one line of conf for each type. Moreover even if we could do it what text do we put? The English version? 
Another solution would be to fix the text for everything except free text but we still have the language issue and we loos e the idea that a user wants to change "Photo taken" in "Crazy photos".
I find it more understandable to let the user put whatever he wants. If someone wants to use its 20x4 to show:

> Photos takens: 465
> Photos prises: 465
> Fotos : 465
> фотографии : 465

He can (I'm not sure the LCD supports Cyrillic alphabet though...)

And about the 

> Transfer the repository to the pibooth organisation.

If you agree I'll see with Anxaue the other pibooth developper if he agrees then I'll move this repo to the organisation.
Note that you'll always have the same access to there repo, it will only change its url from https://github.com/DJ-Dingo/pibooth-lcd-I2c/ to https://github.com/pibooth/pibooth-lcd-I2c/

**Comment by DJ-Dingo** [comment]
Nice :-) 

We would use the english. as standard and then the user can change text - if understand it right
Agree, i also do **not** like the fix text, the idea would go away.

In my version the text for each line of type was in the config, but i can see that it's tricky here becouse we now have a selection in the menu. with mean you only read once and save once on exit. 

What if there was a tick you could check or even better a fast button, (get standard text, for whatever type you selected) out of each line, and it would fast get the standard texts (english), but you can still change the text after it gets the standard text? 
and make it so you inside config.cfg, can make the standard text chance to your country or what ever you want it to be.?
That way it just fast overwrite the text on the line, to your own standard.

Just found out that it wont show Danish characters æøå ÆØÅ thought I allready had tested that part.
I have to look into that, to see why it can't anymore. Maybe i did not used CharLCD back then.

Okay with Transfer

questions
The instalation of py is in home/pi/.local/lib/python3.7/site-packages
And not in the pibooth plugin folder or itś own folder. is there a reason for that, just curious.

On the https://pypi.org/project/pibooth-lcd-i2c/ - at the top it says "**pip** install pibooth-lcd-i2c"
In the install in the readme it say "$ **pip3** install pibooth-lcd-i2c" with i guess is the right install ?.

And a last thing, i will make the normal pibooth_lcd soon (lcd without I2c portexpander) it has some extra things that need to be added with pin, but basic the same py file, and a totally new readme file. I let you know when i am done, an add you as admin to that too. Will still suggest to use the I2c version, but if someone has a lcd lying around without an I2c, and want use it, they can :-)

**Comment by werdeil** [comment]
> What if there was a tick you could check or even better a fast button,

I like the idea of a button to set to default, it could be useful for other part of the menu, I'll discuss it with anxuae.

> The instalation of py is in home/pi/.local/lib/python3.7/site-packages
And not in the pibooth plugin folder or itś own folder. is there a reason for that, just curious.

Yes even if it is a plugin, for python (and pip) it is considered as a standalone package, and installed in the package folder.

> On the https://pypi.org/project/pibooth-lcd-i2c/ - at the top it says "pip install pibooth-lcd-i2c"
In the install in the readme it say "$ pip3 install pibooth-lcd-i2c" with i guess is the right install ?

Yeah pip3 is an alias for pip, it is useful if you have python2 and python3 installed on your machine (and pip2 and pip3) when writing `pip` in the command you don't know which one to use, when you write pip3 you are sure that it use the one linked to python3.

> And a last thing, i will make the normal pibooth_lcd soon (lcd without I2c portexpander) it has some extra things that need to be added with pin, but basic the same py file, and a totally new readme file. I let you know when i am done, an add you as admin to that too. Will still suggest to use the I2c version, but if someone has a lcd lying around without an I2c, and want use it, they can :-)

Yeah I have seen that you had this second plugin. Tell me when it is done, I'm wondering if there is a way to merge them in one.

I'm doing the transfer tonight

**Comment by DJ-Dingo** [comment]
Cool. with transfer.

Cool with the buttons.
Set to default should be for every Line text for every Type right?  - not all the lines at the same time
4 buttons one for each line text. If you use keyboard the button should be out of the text. since we have selection on the Type line which is endless.

Standars could be = Photos Taken, Printed, Forgotten, Duplicates, %d/%m - %H:%M:%S , Your Text...
Offcouse "Printed" could be "Printed Photos" on a 20x4, so being able to chance the standards in the config.cfg would be nice.

Yes i let you know about pibooth_lcd, just wanted to check every details and setups first. Since it is driven by GPIO pins, i allso need to make some standard pins that make sence without using the pins we are using in other plugins. And test it.
On the net there are som standard pins for setup lcd, but they use some of the same as pibooth plugins.

I was thinking the same on marge them, but think we will run in to small problems, since the setup are very different.
And the readme file would be very long. So to not confuse to much it's probably best to keep them seperate, and tell about each one in the top of the readme files.

**Comment by DJ-Dingo** [comment]
Just want you to know. 
On https://pypi.org/project/pibooth-lcd-i2c/ 
it shows under the download pic in the top - downloads package not found

**Comment by werdeil** [comment]
> it shows under the download pic in the top - downloads package not found

It needs some time to update, now it is ok.

About the transfer, I fact I need more rights on the repo, I don't have access to the "settings" menu where there is the option to move the repo.

**Comment by DJ-Dingo** [comment]
yeah ok.

okey, how do i do that?
There is a Danger Zone with transfer

**Comment by werdeil** [comment]
I don't think you can do the transfer either as you are not part of the pibooth organization.

**Comment by DJ-Dingo** [comment]
Hi @werdeil 

As you have probably seen, the [GPIO ](https://github.com/DJ-Dingo/pibooth-lcd-gpio)plugin is complete. Should I try to merge them into one. that way we can also change the name from I2c and [GPIO ](https://github.com/DJ-Dingo/pibooth-lcd-gpio) to just one pibooth-LCD-Display. 

I have also merged my latest plugin [Pibooth-OLED-I2c-SPI](https://github.com/DJ-Dingo/pibooth-oled-i2c-spi) plugin. to use both I2c and SPI displays. (Which you have also been given access to).

**Comment by werdeil** [comment]
Hi @DJ-Dingo I'm sorry I don't really have the time to look into your code as I've change job and I'm moving.

I think indeed it could be a good idea to try to merge both LCD plugin into one, if this doesn't add to much settings, and yes change the name too.

I hope I'll have time to review your changes then and upload a new version of the plugin on pip.

**Comment by DJ-Dingo** [comment]
Hi @werdeil good luck with all that, and Merry Christmas.

I will do that, and make the tests. I let you know when a workin version is ready for you to look at.

**Comment by DJ-Dingo** [comment]
@werdeil Don´t know if you have seen it, I made a Pull requests for the new version 2.0 [https://github.com/DJ-Dingo/pibooth-lcd-I2c/pull/9](https://github.com/DJ-Dingo/pibooth-lcd-I2c/pull/9)

**Comment by werdeil** [comment]
I'll upload the new plugin on pip this week-end (as the name changed it will be a new one I guess).

About moving this repo in the pibooth organization, if you agree of course, you need to give me the admin rights on the plugin so I can do the "transfer ownership" action
<img width="971" alt="image" src="https://user-images.githubusercontent.com/6229844/119987348-37c43480-bfc5-11eb-8ca6-0cdbd971ea0b.png">

**Comment by DJ-Dingo** [comment]
@werdeil I try that, but right now the installation of the plugin no longer work, after PR and i changed the name.
You need to make the changes in the setup file, like (link version etc.)

**Comment by DJ-Dingo** [comment]
@werdeil 
Think it is best that i do the transfer after the new release, and when we both are online. 
Since the new owner have to accept the transfer within one day, or the invitation will expire.

It also says this:
"To transfer a repository that you own to an organization, you must have permission to create a repository in the target organization."

And It says something about potential loss, that i not really understand.
"If you transfer a private repository to a GitHub Free user or organization account, the repository will lose access to features like protected branches and GitHub Pages."

**Comment by DJ-Dingo** [comment]
@werdeil 
I tryed to chance you from collaborator to admin, but it don´t give me the same options, as it says in the Docs https://docs.github.com/

UPDATE: But for user accounts collaborators have the same rigths, you can not give different permissions unles you have an organization account.

I try do the transfer when we are both online.

**Comment by werdeil** [comment]
Release finally done (sorry for the delay)
https://pypi.org/project/pibooth-lcd-display/2.0.0/
I've added a release.rst page to have reminder on how to do it (copied from pibooth). If you have a pypi acount I can send you invite via pypi to allow you full rights on it.

**Comment by DJ-Dingo** [comment]
Nice @werdeil - I test it later today
(But It still says "Releases 1 v1.0.0 Latest on 6 Nov 2020") on the right side.

Yes i have pypi https://pypi.org/user/DJ-Dingo/

And when you have the time, lets have and install for the small cool "OLED screens" as well. (I will chance the name in the weekend, to just "pibooth-oled, or pibooth-oled-screen or pibooth-oled-display" Which name do you prefer?. 
https://github.com/DJ-Dingo/pibooth-oled-i2c-spi

And also lets see if we can do the transfer soon too.

**Comment by werdeil** [comment]
I forgot to create the release on GitHub, now it is done :-) and I've added you as owner on the pypi project.

For the "pibooth-oled-display" (my preference as it is not used as a screen for the pibooth but an extra display) I won't be available in the next weeks, maybe in August.

**Comment by DJ-Dingo** [comment]
Cool.

I chance to "pibooth-oled-display" then. August is fine, I might have a few more ideas I want to add before that :-)
